src/main/java/de/rwth/idsg/ocpp/jaxb/JavaDateTimeConverter.java

```diff
@@ -0,0 +1,90 @@
+package de.rwth.idsg.ocpp.jaxb;
+
+import jakarta.xml.bind.annotation.adapters.XmlAdapter;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalAccessor;
+import java.util.Locale;
+
+/**
+ * Java-Time and XSD represent data and time information according to ISO 8601.
+ *
+ * @author Sevket Goekay <goekay@dbis.rwth-aachen.de>
+ * @since 20.10.2014
+ */
+public class JavaDateTimeConverter extends XmlAdapter<String, OffsetDateTime> {
+
+    // Flexible ISO 8601 parser: supports the optional fraction and optional offset
+    private static final DateTimeFormatter ISO_PARSER = new DateTimeFormatterBuilder()
+        // Date: yyyy-MM-dd
+        .appendValue(ChronoField.YEAR, 4, 10, java.time.format.SignStyle.EXCEEDS_PAD)
+        .appendLiteral('-')
+        .appendValue(ChronoField.MONTH_OF_YEAR, 2)
+        .appendLiteral('-')
+        .appendValue(ChronoField.DAY_OF_MONTH, 2)
+
+        // Time: 'T'HH:mm:ss
+        .appendLiteral('T')
+        .appendValue(ChronoField.HOUR_OF_DAY, 2)
+        .appendLiteral(':')
+        .appendValue(ChronoField.MINUTE_OF_HOUR, 2)
+        .appendLiteral(':')
+        .appendValue(ChronoField.SECOND_OF_MINUTE, 2)
+
+        // Optional: .SSS... (fractional seconds)
+        .optionalStart()
+        .appendLiteral('.')
+        .appendFraction(ChronoField.NANO_OF_SECOND, 1, 9, false)
+        .optionalEnd()
+
+        // Optional: +02:00 or Z (offset)
+        .optionalStart()
+        .appendOffsetId()
+        .optionalEnd()
+
+        .toFormatter(Locale.ROOT);
+
+    // Outputs UTC timestamps with millisecond precision and trailing 'Z'
+    private static final DateTimeFormatter UTC_FORMATTER = new DateTimeFormatterBuilder()
+        .appendInstant(3)
+        .toFormatter(Locale.ROOT);
+
+    @Override
+    public OffsetDateTime unmarshal(String v) throws Exception {
+        if (isNullOrEmpty(v)) {
+            return null;
+        } else {
+            TemporalAccessor parsed = ISO_PARSER.parse(v);
+            if (parsed.isSupported(ChronoField.OFFSET_SECONDS)) {
+                // Has explicit offset → safe to parse as OffsetDateTime
+                return OffsetDateTime.from(parsed).withOffsetSameInstant(ZoneOffset.UTC);
+            } else {
+                // No offset → treat as UTC
+                LocalDateTime ldt = LocalDateTime.from(parsed);
+                return ldt.atOffset(ZoneOffset.UTC);
+            }
```

Comment:
@goekay 

> since some weird charging station implementations do NOT send timezone

The issue you mention is fixed here and is validated by the tests.